python/foglamp/common/configuration_manager.py

```diff
@@ -265,13 +278,22 @@ def get_entry_val(k):
 
     async def _create_new_category(self, category_name, category_val, category_description):
         try:
+            if isinstance(category_val, dict):
+                new_category_val = copy.deepcopy(category_val)
+                # Remove "deprecated" items from a new category configuration
+                for i, v in category_val.items():
+                    if 'deprecated' in v:
+                        new_category_val.pop(i)
```

Comment:
I think we need an Audit log here for the removal of deprecated

Reply:
First of all, this is _create_new_category() method. A check for "deprecated" attribute is required here to prevent the deprecated items from going into configuration right at the very first time. Secondly, removing "deprecated" items in _merge_category_vals() is identical to removing an  item per the regular merge process logic. Since _merge_category_vals() method never adds an Audit log entry for its operations, we also choose not to do so for removing the deprecated items.